js/adapt-assessmentArticleModel.js

```diff
@@ -132,12 +132,82 @@ define([
                 }
             }
 
+            var questionBanksEnabled = this.get('_assessment')._banks._isEnabled;
+            var questionBanksLoaded = this._questionBanks;
+
             if (!quizModels) {
-                // leave the order as before, completed or not
-                quizModels = this.getChildren().models;
+                if (questionBanksEnabled && !questionBanksLoaded) {
+
+                  this.getChildren().models = this._originalChildModels;
+                  if(assessmentConfig._banks &&
+                          assessmentConfig._banks._isEnabled &&
+                          assessmentConfig._banks._split.length > 1) {
+
+                      quizModels = this._setupBankedAssessment();
+                  } else if(assessmentConfig._randomisation &&
+                          assessmentConfig._randomisation._isEnabled) {
```

Comment:
Is this _banks randomisation or randomised blocks? If it's the later it is not dependent on `questionBanksEnabled && !questionBanksLoaded` and looks like a duplicate of the code below in the else statement.

Reply:
It is randomised blocks so i've added a check for that in the if statement.